tests/framework_starlette/test_bg_tasks.py

```diff
@@ -83,34 +74,44 @@ def _test():
 @skip_if_no_middleware
 @pytest.mark.parametrize("route", ["async", "sync"])
 def test_basehttp_style_middleware(target_application, route):
-    route_metrics = [("Function/_test_bg_tasks:run_%s_bg_task" % route, 1)]
-    old_metrics = [
-        ("Function/_test_bg_tasks:%s_bg_task" % route, 1),
-        ("Function/_test_bg_tasks:run_%s_bg_task" % route, 1),
-    ]
+    route_metric = ("Function/_test_bg_tasks:run_%s_bg_task" % route, 1)
+    # A function trace metric that appears only when the bug below is present, causing background tasks to be
+    # completed inside web transactions, requiring a function trace to be used for timing
+    # instead of a background task transaction. Should not be present at all when bug is fixed.
+    bg_task_metric = ("Function/_test_bg_tasks:%s_bg_task" % route, 1)
 
     def _test():
         app = target_application["basehttp"]
         response = app.get("/" + route)
         assert response.status == 200
 
-    if starlette_version >= (0, 20, 1):
-        if sys.version_info[:2] > (3, 7):
-            _test = validate_transaction_metrics(
-                "_test_bg_tasks:run_%s_bg_task" % route, index=-2, scoped_metrics=route_metrics
-            )(_test)
-            _test = validate_transaction_metrics(
-                "_test_bg_tasks:%s_bg_task" % route, background_task=True
-            )(_test)
-            _test = validate_transaction_count(2)(_test)
-        else:   # Python <= 3.7 requires this specific configuration with starlette 0.20.1
-            _test = validate_transaction_metrics(
-                "_test_bg_tasks:run_%s_bg_task" % route, scoped_metrics=route_metrics
-            )(_test)
-            _test = validate_transaction_count(1)(_test)
+    BUG_COMPLETELY_FIXED = (starlette_version >= (0, 21, 0)) or (
+        starlette_version >= (0, 20, 1) and sys.version_info[:2] > (3, 7)
+    )
+    BUG_PARTIALLY_FIXED = (
+        starlette_version >= (0, 20, 1) and starlette_version < (0, 21, 0) and sys.version_info[:2] <= (3, 7)
```

Comment:
I suggest the following instead as it's a lot more readable:
```suggestion
       (0, 20, 1) <= starlette_version < (0, 21, 0) and sys.version_info[:2] <= (3, 7)
```